events/loginEvent.py

```diff
@@ -90,6 +93,21 @@ async def handle(web_handler: AsyncRequestHandler) -> tuple[str, bytes]:  # toke
             # Invalid password
             raise exceptions.loginFailedException()
 
+        try:
+            # we have a user ID we can rely on, allow further processing of login body
+            bancho_login_request = {
+                "login_body": web_handler.request.body.decode(),
+                "user_id": userID,
+            }
+
+            for routing_key in settings.BANCHO_LOGIN_ROUTING_KEYS:
+                await glob.amqp_channel.default_exchange.publish(
+                    aio_pika.Message(body=orjson.dumps(bancho_login_request)),
```

Comment:
can we parse the message on bancho-service & emit something more generic uncoupled from the bancho request format?

Reply:
we *could*, but i'm intentionally trying to obfuscate what this could be doing (the consumer for this is doing quite a bit of parsing itself)

Reply:
also to make that work well we'd probably have to rewrite a large part of the logic of this function just generally so that i can even create some kind of object. it's so tangled that IDE can't even analyse. not sure it's a good idea

Reply:
What about copy/pasting this into the try/catch and using it for the data sent over aiopika? (not refactoring any other code around here, just this step for now)

https://github.com/cmyui/bancho.py/blob/06495df993e3fbc8750bed2e7a365f6088b9804e/app/api/domains/cho.py#L502-L538

Reply:
will have to rewrite all of my consumer, but sure

Reply:
> (the consumer for this is doing quite a bit of parsing itself)

I think in general this is a bad pattern & I don't suspect this ^ suggestion hinders obfuscation/security concerns

Reply:
i don't like it as a pattern either but it came from wanting to obfuscate intent & not wanting to touch anymore code than necessary

Reply:
https://github.com/osuAkatsuki/bancho-service/pull/188/commits/208977789c3a95c8844dba23cf4b9b6ea21af2af

Reply:
Since we've already auth'd by this point (above), I think we should strip password_md5 out of the msg when sending over AMQP as well

Reply:
sure

Reply:
maybe `bancho_login_request` -> `amqp_login_message` naming for clarity not to reuse it later, then it can be mutated inplace

Reply:
https://github.com/osuAkatsuki/bancho-service/pull/188/commits/e1e20c683a2b7c932cacf14c3a5d205cac5afcd2